File: c/Matrizes/soma_linhas.c
    #include <stdio.h>

int main() {

    int M, N;
    double matriz[10][10];
    double somaLinhas[10];

    printf("Quantas linhas tera a matriz? ");
    scanf("%d", &M);

    printf("Quantas colunas tera a matriz? ");
    scanf("%d", &N);

    for (int i = 0; i < M; i++) {
        for (int j = 0; j < N; j++) {

            printf("Elemento [%d,%d]: ", i, j);
            scanf("%lf", &matriz[i][j]);
        }
    }

    for (int i = 0; i < M; i++) {

        somaLinhas[i] = 0;

        for (int j = 0; j < N; j++) {
            somaLinhas[i] += matriz[i][j];
        }
    }

    printf("\nVETOR GERADO:\n");

    for (int i = 0; i < M; i++) {
        printf("%.1lf\n", somaLinhas[i]);
    }

    return 0;
}

File: java/Matrizes/SomaLinhas.java
import java.util.Locale;
import java.util.Scanner;

public class SomaLinhas {
    public static void main(String[] args) {

        Locale.setDefault(Locale.US);
        Scanner sc = new Scanner(System.in);

        int M, N;

        System.out.print("Quantas linhas? ");
        M = sc.nextInt();

        System.out.print("Quantas colunas? ");
        N = sc.nextInt();

        double[][] matriz = new double[M][N];
        double[] vetor = new double[M];

        // Leitura da matriz
        for (int i = 0; i < M; i++) {
            for (int j = 0; j < N; j++) {
                System.out.print("Elemento [" + i + "," + j + "]: ");
                matriz[i][j] = sc.nextDouble();
            }
        }

        // Soma das linhas
        for (int i = 0; i < M; i++) {

            double soma = 0;

            for (int j = 0; j < N; j++) {
                soma += matriz[i][j];
            }

            vetor[i] = soma;
        }

        System.out.println("Vetor gerado:");

        for (int i = 0; i < M; i++) {
            System.out.printf("%.1f%n", vetor[i]);
        }

        sc.close();
    }
}